File: programmers/Level. 2/올바른 괄호.py
# 프로그래머스 Lv.2 올바른 괄호

def solution(s):
    check = 0
    
    # 스택을 활용
    for i in s :
        if i == '(' :
            check += 1
        elif i == ')' :
            check -= 1
        
        if check < 0 :
            return False
    
    # 0일시 True, 아닐시 False 리턴
    return check == 0

File: programmers/Level. 2/올바른 괄호.cpp
// 프로그래머스 Lv.2 올바른 괄호

#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int check = 0;
    
    for (int i = 0; i < s.size(); i++) {
        if (s[i] == '(')
            check++;
        else if (s[i] == ')')
            check--;
        
        if (check < 0)
            return false;
    }
    
    return check == 0;
}